roachtest: acceptance/version-upgrade failed

[(roachtest).acceptance/version-upgrade failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2187197&tab=buildLog) on [master@edb904bb60f18ec6dbec85a42701e96781ad40e2](https://github.com/cockroachdb/cockroach/commits/edb904bb60f18ec6dbec85a42701e96781ad40e2):

```
		  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:1808
		  |   | runtime.main
		  |   | 	/usr/local/go/src/runtime/proc.go:203
		  |   | runtime.goexit
		  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1357
		  | Wraps: (2) 3 safe details enclosed
		  | Wraps: (3) 2: dead
		  | Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *errors.errorString
		Wraps: (3) secondary error attachment
		  | 3: dead
		  | (1) attached stack trace
		  |   | main.glob..func13
		  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:1115
		  |   | main.wrap.func1
		  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:266
		  |   | github.com/spf13/cobra.(*Command).execute
		  |   | 	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830
		  |   | github.com/spf13/cobra.(*Command).ExecuteC
		  |   | 	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
		  |   | github.com/spf13/cobra.(*Command).Execute
		  |   | 	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
		  |   | main.main
		  |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:1808
		  |   | runtime.main
		  |   | 	/usr/local/go/src/runtime/proc.go:203
		  |   | runtime.goexit
		  |   | 	/usr/local/go/src/runtime/asm_amd64.s:1357
		  | Wraps: (2) 3 safe details enclosed
		  | Wraps: (3) 3: dead
		  | Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *errors.errorString
		Wraps: (4) attached stack trace
		  | main.glob..func13
		  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:1115
		  | main.wrap.func1
		  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:266
		  | github.com/spf13/cobra.(*Command).execute
		  | 	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830
		  | github.com/spf13/cobra.(*Command).ExecuteC
		  | 	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
		  | github.com/spf13/cobra.(*Command).Execute
		  | 	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
		  | main.main
		  | 	/go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:1808
		  | runtime.main
		  | 	/usr/local/go/src/runtime/proc.go:203
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1357
		Wraps: (5) 3 safe details enclosed
		Wraps: (6) 4: dead
		Error types: (1) errors.Unclassified (2) *secondary.withSecondaryError (3) *secondary.withSecondaryError (4) *withstack.withStack (5) *safedetails.withSafeDetails (6) *errors.errorString

```

<details><summary>More</summary><p>


Artifacts: [/acceptance/version-upgrade](https://teamcity.cockroachdb.com/viewLog.html?buildId=2187197&tab=artifacts#/acceptance/version-upgrade)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aacceptance%2Fversion-upgrade.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

```
panic: unexpected Family: Box2DFamily

goroutine 7547 [running]:
panic(0x4aa21e0, 0xc000785160)
	/usr/local/go/src/runtime/panic.go:722 +0x2c2 fp=0xc0017f56e8 sp=0xc0017f5658 pc=0x763fd2
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).SQLStandardNameWithTypmod(0xc0021bb2c0, 0x0, 0x0, 0x1, 0xc0030b8b08)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1573 +0x1781 fp=0xc0017f5860 sp=0xc0017f56e8 pc=0x15f5c51
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).SQLStandardName(0xc0021bb2c0, 0xc0030b8b08, 0x2e)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1390 +0x39 fp=0xc0017f5898 sp=0xc0017f5860 pc=0x15f4419
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).InformationSchemaName(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1587
github.com/cockroachdb/cockroach/pkg/sql.glob..func67.1.1(0xc001af9b00, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:411 +0x11d8 fp=0xc0017f5b28 sp=0xc0017f5898 pc=0x31c5028
github.com/cockroachdb/cockroach/pkg/sql.forEachColumnInTable(0xc001f24f00, 0xc0017f5bb8, 0x0, 0x73dd7b)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1821 +0x83 fp=0xc0017f5b60 sp=0xc0017f5b28 pc=0x30c2483
github.com/cockroachdb/cockroach/pkg/sql.glob..func67.1(0xc003193ec0, 0x4e9a7da, 0x6, 0xc001f24f00, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:378 +0x166 fp=0xc0017f5c08 sp=0xc0017f5b60 pc=0x31c54a6
github.com/cockroachdb/cockroach/pkg/sql.forEachTableDesc.func1(0xc003193ec0, 0x4e9a7da, 0x6, 0xc001f24f00, 0xc000b9e540, 0x0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1633 +0x4d fp=0xc0017f5c48 sp=0xc0017f5c08 pc=0x319f2ad
github.com/cockroachdb/cockroach/pkg/sql.forEachTableDescWithTableLookupInternal(0x5f76080, 0xc002288b80, 0xc002017170, 0xc001850600, 0x0, 0x4b28400, 0xc0017f5d88, 0xc0024ffb00, 0x60)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1793 +0x421 fp=0xc0017f5d40 sp=0xc0017f5c48 pc=0x30c2051
github.com/cockroachdb/cockroach/pkg/sql.forEachTableDescWithTableLookup(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1698
github.com/cockroachdb/cockroach/pkg/sql.forEachTableDesc(0x5f76080, 0xc002288b80, 0xc002017170, 0xc001850600, 0x0, 0xc0015e15e8, 0x0, 0xc0015e1610)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1627 +0x8d fp=0xc0017f5da8 sp=0xc0017f5d40 pc=0x30c183d
github.com/cockroachdb/cockroach/pkg/sql.glob..func67(0x5f76080, 0xc002288b80, 0xc002017170, 0xc001850600, 0xc001850660, 0xc0015e1780, 0x791b60)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:373 +0xa7 fp=0xc0017f5e20 sp=0xc0017f5da8 pc=0x317e0a7
github.com/cockroachdb/cockroach/pkg/sql.virtualDefEntry.getPlanInfo.func1.1(0x5edff40, 0xc002dd84c0, 0x2, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:484 +0x11a fp=0xc0017f5e98 sp=0xc0017f5e20 pc=0x31b537a
github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func2(0x5f76080, 0xc002288d40, 0xc00369d320, 0xc00287b500, 0xc002dd84c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:112 +0x109 fp=0xc0017f5fb8 sp=0xc0017f5e98 pc=0x31b6fd9
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc0017f5fc0 sp=0xc0017f5fb8 pc=0x7947a1
created by github.com/cockroachdb/cockroach/pkg/sql.setupGenerator
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:101 +0xf7
```

```
panic: unexpected Family: 25 [recovered]
	panic: panic while executing 1 statements: SELECT _ FROM [SHOW COLUMNS FROM _] ORDER BY random() LIMIT _; caused by unexpected Family: 25
```

Also seeing the above in the most recent failure.

--- a/pkg/cmd/roachtest/acceptance.go
+++ b/pkg/cmd/roachtest/acceptance.go
@@ -64,7 +64,6 @@ func registerAcceptance(r *testRegistry) {
 			// to head after 19.2 fails.
 			minVersion: "v19.2.0",
 			timeout:    30 * time.Minute,
-			skip:       "https://github.com/cockroachdb/cockroach/issues/52907",
 		},
 	}
 	tags := []string{"default", "quick"}

--- a/pkg/kv/kvserver/batcheval/cmd_end_transaction.go
+++ b/pkg/kv/kvserver/batcheval/cmd_end_transaction.go
@@ -904,6 +904,15 @@ func splitTriggerHelper(
 		return enginepb.MVCCStats{}, result.Result{}, err
 	}
 
+	if !rec.ClusterSettings().Version.IsActive(ctx, clusterversion.VersionAbortSpanBytes) {
+		// Since the stats here is used to seed the initial state for the RHS
+		// replicas, we need to be careful about zero-ing out the abort span
+		// bytes if the cluster version introducing it is not yet active. Not
+		// doing so can result in inconsistencies in MVCCStats across replicas
+		// in a mixed-version cluster.
+		h.AbsPostSplitRight().AbortSpanBytes = 0
+	}
+
 	// Note: we don't copy the queue last processed times. This means
 	// we'll process the RHS range in consistency and time series
 	// maintenance queues again possibly sooner than if we copied. The
@@ -1032,14 +1041,6 @@ func splitTriggerHelper(
 	deltaPostSplitLeft := h.DeltaPostSplitLeft()
 	if !rec.ClusterSettings().Version.IsActive(ctx, clusterversion.VersionContainsEstimatesCounter) {
 		deltaPostSplitLeft.ContainsEstimates = 0
-	}
-	if !rec.ClusterSettings().Version.IsActive(ctx, clusterversion.VersionAbortSpanBytes) {
-		// Since the stats here is used to seed the initial state for the RHS
-		// replicas, we need to be careful about zero-ing out the abort span
-		// bytes if the cluster version introducing it is not yet active. Not
-		// doing so can result in inconsistencies in MVCCStats across replicas
-		// in a mixed-version cluster.
-		pd.Replicated.Split.RHSDelta.AbortSpanBytes = 0
 	}
 	return deltaPostSplitLeft, pd, nil
 }

